queue.go

```diff
@@ -165,6 +166,7 @@ func (q *Queue) Put(val interface{}, timeout float64) error {
 	q.clearPending()
 	isfull := q.isfull()
 	if timeout < 0.0 && isfull {
+		q.mutex.Unlock()
```

Comment:
But your every put/get operation have lock operation and 
```
	if timeout < 0.0 && isempty {
		return nil, ErrEmptyQueue
	}
```

this exit case not unlock.